.github/workflows/reusable-build-test-release.yml

```diff
@@ -55,94 +55,98 @@ on:
-      - name: skip workflow if description is empty for labeled pr
-        id: skip-workflow
-        env:
-          PR_BODY: ${{ github.event.pull_request.body }}
-        run: |
-          set +e
-          TESTSET="knowledge ui modinput_functional scripted_inputs escu requirement_test"
-          echo "testset=$TESTSET" >> "$GITHUB_OUTPUT"
-          SKIP_WORKFLOW="No"
-          if [[ '${{ github.event.action }}' == 'labeled' && '${{ github.event.label.name }}' == 'preserve_infra' ]]; then
-            echo "$PR_BODY" >> body.txt
-            SKIP_WORKFLOW="Yes" 
-            tests=$(grep -i "^preserve:" body.txt | { grep -v grep || true; })
-            for test_type in $TESTSET; do
-                if [[ $tests =~ $test_type ]]; then
-                    SKIP_WORKFLOW="No"
-                fi
-            done
-          fi
-          echo "skip-workflow=$SKIP_WORKFLOW" >> "$GITHUB_OUTPUT"
-          if [ "$SKIP_WORKFLOW" == "Yes" ]; then 
-            echo "No description is provided with preserve infra label"
-          fi 
-      - name: setup for delay destroy
-        id: delay-destroy-setup
-        shell: bash
-        env:
-          PR_BODY: ${{ github.event.pull_request.body }}
+      - name: Configure Test Execution
+        id: configure-tests
         run: |
           set +e
-          TESTSET="${{ steps.skip-workflow.outputs.testset }}"
-          for test_type in $TESTSET; do
-              eval DELAY_DESTROY_"$test_type"="No"
-              eval EXECUTE_"$test_type"="Yes"
+          declare -A EXECUTE
+          TESTSET=("knowledge" "ui" "modinput" "scripted_inputs" "escu" "requirement")
+          delay_destroy="No"
+
+          for test_type in "${TESTSET[@]}"; do
+            EXECUTE["$test_type"]="false"
           done
-          if [[ '${{ github.event.label.name }}' == 'preserve_infra' ]]; then
-            echo "$PR_BODY" >> body.txt
-            tests=$(grep -i "^preserve:" body.txt | { grep -v grep || true; })
-            if [[ $tests =~ "escu" ]]; then
-              echo "preserve_infra for escu test-type is not supported yet"
-            fi
-            for test_type in $TESTSET; do
-                if [[ $tests =~ $test_type ]]; then
-                    eval DELAY_DESTROY_"$test_type"="Yes"
-                else
-                    eval EXECUTE_"$test_type"="No" 
+
+          case "${{ github.event_name }}" in
+            "pull_request")
+              if ${{ contains(github.event.pull_request.labels.*.name, 'preserve-infra') }}; then
+                delay_destroy="Yes"
+              fi
+              if ${{ github.base_ref == 'main' }}; then
+                for test_type in "${TESTSET[@]}"; do
+                  EXECUTE["$test_type"]="true"
+                done
+              else
+                labels=$(echo '${{ toJSON(github.event.pull_request.labels) }}' | jq -r '.[] | .name')
+                for test_type in "${TESTSET[@]}"; do
+                  if [[ "$labels" =~ "$test_type" ]]; then
```

Comment:
This is the line which handles label logic, so if I put a label `knowledge`, it means that we are going to run only `knowledge` tests?

Reply:
So, if the pipeline is running and I update the PR with an additional label it will trigger the pipeline again?

Reply:
Let's document this behaviour and trust people to use it accordingly.